client/src/redux/challenges-info-saga.js

```diff
@@ -0,0 +1,84 @@
+import { select, put, takeEvery } from 'redux-saga/effects';
+
+import { SuperBlocks } from '../../../shared/config/curriculum';
+import superBlockStructure from '../../../curriculum/superblock-structure/full-stack.json';
+import {
+  challengeMetaSelector,
+  completedChallengesIdsSelector
+} from '../templates/Challenges/redux/selectors';
+import { updateAllChallengesInfoWithCompletionState } from './actions';
+
+const getCompletionState = ({
```

Comment:
At the time of the challenge submission, we should know the challenge meta (challenge id, block, module, chapter, superblock). 

Since we also know the completedChallenges array, and the complete structure of the curriculum ( which challenges belongs to which block, challenge, module, chapter, superblock), can't we check if the all of the blocks that belong to the current module are completed?

If so, we can reuse some of the isBlockNewlyCompletedSelector selector and we don't have to push completion states to the store.

Does that make sense?